src/inmanta/server/services/orchestrationservice.py

```diff
@@ -916,13 +924,7 @@ async def put_version(
                     pip_config=pip_config,
                     connection=con,
                 )
-            try:
-                await self._trigger_auto_deploy(env, version, connection=con)
-            except Conflict as e:
-                # this should be an api warning, but this is not supported here
-                LOGGER.warning(
-                    "Could not perform auto deploy on version %d in environment %s, because %s", version, env.id, e.log_message
-                )
+            await self._trigger_auto_deploy(env, version)
```

Comment:
I think we need to add to a comment a few lines up where we call `get_connection` that it must never be changed to use an existing, encapsulating connection object because we trigger a background task that requires this method's changes to be committed. i.e. avoid races like the ones we introduced in #6955.